src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -136,66 +109,106 @@ private enum BatchWriterState
             Error
         }
 
+        /// <summary>
+        /// >The batch-specific URL converter that stores the content IDs found in a changeset and supports resolving cross-referencing URLs.
+        /// </summary>
+        internal ODataBatchPayloadUriConverter PayloadUriConverter
+        {
+            get { return this.payloadUriConverter; }
+        }
+
+        /// <summary>The message for the operation that is currently written; or null if no operation is written right now.</summary>
+        internal ODataBatchOperationMessage CurrentOperationMessage

```

Comment:
>internal [](start = 8, length = 8)

This should be private, not internal.  Derived classes should access the current method through either CurrentOperationRequestMessage or CurrentOperationResponseMessage.  If that can't be done, it should be protected, not internal.

Reply:
Per offline discussion, remove this CurrentOperationMessage code block from base class, and duplicate(unfortunately) it into both derived classes as private. Keep the ODataBatchOperationMessage type internal for now, ODL-external implementation of batch writer either provide associate implementation itself or ask ODL to make ODataBatchOperationMessage, etc public.

---
In reply to: [141129222](https://github.com/OData/odata.net/pull/960#discussion_r141129222) [](ancestors = 141129222)

Reply:
It looks like you split this into CurrentOperationRequestMessage and CurrentOperationResponseMessage, which return the (already public) ODataBatchOperation[Request|Response]Message, which makes a lot more sense.

---
In reply to: [143308744](https://github.com/OData/odata.net/pull/960#discussion_r143308744) [](ancestors = 143308744,141129222)